zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/ProcessInstanceStateTransitionGuard.java

```diff
@@ -187,4 +200,149 @@ private Either<String, ElementInstance> hasNonInterruptedFlowScope(
+      final BpmnElementContext context, final ExecutableFlowElement executableFlowElement) {
+    final Optional<DeployedProcess> deployedProcess =
+        stateBehavior.getProcess(context.getProcessDefinitionKey(), context.getTenantId());
+
+    return deployedProcess
+        .map(DeployedProcess::getProcess)
+        .map(process -> doesPathExist(context, executableFlowElement, process))
+        .orElse(
+            Either.left(
+                String.format(
+                    "Expected to find a deployed process for process definition key '%d', but none found.",
+                    context.getProcessDefinitionKey())));
+  }
+
+  private Either<String, ?> doesPathExist(
+      final BpmnElementContext context,
+      final ExecutableFlowElement executableFlowElement,
+      final ExecutableProcess process) {
+    final var targetElementId = executableFlowElement.getId();
+    final var activateElementIds = findActivateElementInFlowScope(context, targetElementId);
+    final Optional<DirectBuffer> exists =
+        activateElementIds.stream()
+            .filter(
+                elementId -> doesPathExist(process, elementId, targetElementId, new HashSet<>()))
+            .findAny();
+
+    if (exists.isPresent()) {
+      return Either.left(
+          String.format(
+              "Expected to be able to activate inclusive gateway '%s',"
+                  + " but not all satisfied sequence flows have been taken",
+              BufferUtil.bufferAsString(targetElementId)));
+    } else {
+      return findPendingActivateElementInFlowScope(context, process, targetElementId);
+    }
+  }
+
+  public Set<DirectBuffer> findActivateElementInFlowScope(
+      final BpmnElementContext context, final DirectBuffer targetElementId) {
+    final var flowScopeContext = stateBehavior.getFlowScopeContext(context);
+    final var childInstances = stateBehavior.getChildInstances(flowScopeContext);
+
+    // activate element id in flow scope
+    return childInstances.stream()
+        .map(BpmnElementContext::getElementId)
+        .filter(elementId -> !elementId.equals(targetElementId))
+        .collect(Collectors.toSet());
+  }
+
+  public Either<String, ?> findPendingActivateElementInFlowScope(
+      final BpmnElementContext context,
+      final ExecutableProcess process,
+      final DirectBuffer targetElementId) {
+    final var pendingSequenceFlowIds = stateBehavior.getPendingSequenceFlowIds(context);
+    final var existsPathPendingSequenceFlowIds =
+        pendingSequenceFlowIds.stream()
+            .filter(
+                flowId -> {
+                  final DirectBuffer elementId =
+                      process
+                          .getElementById(flowId, ExecutableSequenceFlow.class)
+                          .getTarget()
+                          .getId();
+
+                  return doesPathExist(process, elementId, targetElementId, new HashSet<>());
+                })
+            .collect(Collectors.toSet());
```

Comment:
:wrench: We should exclude paths ending on an incoming sequence flow of the gateway that was already taken. All incoming flows need to be taken only once.